sources/Context.h

```diff
@@ -146,7 +146,7 @@ class Context {
    * @param cbk the callback to register
    */
   void onNodeOnMainConnexComponent(ProcessNodeCallBackMainComponent &&cbk) {
-    callbacksMainConnexComponent_.push_back(std::forward<ProcessNodeCallBackMainComponent>(cbk));
```

Comment:
I'm quite sure we need to use forward here. Anyway this is not related so please revert this change.

Reply:
ProcessNodeCallBackMainComponent&& isn't a universal reference. It's a rvalue reference so I thought std::move was more appropriate.

Reply:
yes it is a universal reference to a rvalue within a template so forward is required.

Reply:
std::function<void(const std::shared_ptr<inputs::Node> &, std::shared_ptr<dfl::algo::AlgorithmsResults> &)> isn't a template, right ?

Reply:
we can discuss this later on case by case. Anyway this is not related to this development so please revert this in the context of this PR.